### ISSUE TITLE
Rethink operation of grid-analyzer

### ISSUE THREAD

**Issue by tkornut** [IBM/mi-prometheus#66]
Assuming we run:
- grid-trainer that resulted in a grid of trained models (let's say 6 exp = 2 models x 3 problems)
- then we run grid-tester with two runs (with different random seeds of course)

That results in 10 "experiments" that should potentially form the content of a single csv file - 12 rows?

The goal of this issue is to discuss what, how and when should be copied taking into account:
 - we got best_model.pt checkpoint, which indicates in which episode the model was created
 - we got two types of trainers, that could save the model depending on loss calculated on:
      - online-trainer: partial validation set, that is activated every episode % "partial_validation_interval" 
      - offline-trainer: full validation set, that is calculated at the end of every epoch

Besides, offline-trainer can optionally store partial_validation_statistics in csv.file, when one set partial_validation_interval > 0 in config:validation section.

As a result, different trainers will produce different statistics, how to deal with that diversity - this is the goal of that discussion/issue.

**Comment by tkornut** [comment]
Idea:
- save model always using aggregated validation stats (even when it is partial validation)
- online trainer:
    - partial_valid_collector -> export(col) + aggregate -> export(agg) + save model(agg)
    - end: full valid -> export(agg) + save model (agg)
- offline trainer:
    - partial_valid -> export(col)
    - at the end of epoch --> full valid -> export(agg) + save model (agg)
    - end: full valid -> export(agg) + save model (agg)

As a result, we can depend always of the content of validation_statistics_agg.csv for validation data...
i.e. simply read it from model checkpoint!

**Comment by tkornut** [comment]
For Testing: use test_statistics_agg.cvs - this one is easy, we want the aggregated for whole training set, there is only one row in that file.

For training: use training_statistics.csv - we want to use collected (not aggregated) statistics, we can read episode from checkpoint (this is the best model), and find adequate row.

**Comment by tkornut** [comment]
For training: use training_statistics.csv - we want to use collected (not aggregated) statistics, we can read episode from checkpoint (this is the best model), and find adequate row.

**Comment by tkornut** [comment]
As a result, the ONLY thing that we want to really hardcode to read from checkpoint is... episode, which we know that always exists! 👍

**Comment by tkornut** [comment]
Next issue: extraction of timestamps.

We can extract:
 - training timestamp (from directory, indicates when training was started)
 - model timestamp (from checkpoint, indicates when we saved the best model)
 - test timestamp (from test dir, indicates when test  was started)

**Comment by tkornut** [comment]
When it comes to collecting everything in one row, we need to add prefixes to the headers:
 - add "train_" for all columns read from training_statistics.csv,
 - add "valid_" for all columns read from checkpoint.
 - add "test_" for all columns read from test_statistics_agg.csv

**Comment by tkornut** [comment]
As agg statistics for a single batch make no sense, we discussed that with @vmarois and agreed that: 

    online trainer:
        partial_valid_collector -> export(col) + save model (training_col, partial_valid_col)
        end: full valid -> export(agg)  (no save!)
    offline trainer:
        partial_valid -> export(col)
        at the end of epoch --> full valid -> export(agg) + save model (training_agg, full_valid_agg)
        end: full valid -> export(agg) (no save!)

As a result, we can depend always of the content of validation_statistics_agg.csv for validation data...
or even not - simply read the RIGHT ROW with validation statistics aggregated from model checkpoint!

this change is needed to standardize results in grid-analyzer
